modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBColor.java

```diff
@@ -21,43 +21,81 @@
 import org.geotools.mbstyle.parse.MBObjectParser;
 import org.json.simple.JSONArray;
 import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Function;
+
+import java.awt.*;
+import java.util.ArrayList;
 
 /**
- *
+ * A class to transform mapbox color expressions into geotools expressions.
  */
 public class MBColor extends MBExpression{
+    MBObjectParser parse;
 
     public MBColor(JSONArray json) {
         super(json);
+        parse = new MBObjectParser(MBExpression.class);
     }
-
     /**
      * Creates a color value from red, green, and blue components, which must range between 0 and 255,
      * and an alpha component of 1. If any component is out of range, the expression is an error.
      * Example: ["rgb", number, number, number]: color
      * @return
      */
     public Expression colorRGB() {
-            return null;
+        if (json.size() == 4) {
+
+            ArrayList<Expression> rgb = new ArrayList<>();
+            rgb.add(ff.literal("#"));
+
+            Expression er = parse.string(json, 1);
+            er = transformLiteral(er);
+            Function roundr = ff.function("round_2", er);
+            Function hexr = ff.function("tohex", roundr);
+
+            rgb.add(hexr);
+
+            Expression eg = parse.string(json, 2);
+            eg = transformLiteral(eg);
+            Function roundg = ff.function("round_2", eg);
+            Function hexg = ff.function("tohex", roundg);
+            rgb.add(hexg);
+
+            Expression eb = parse.string(json, 3);
+            eb = transformLiteral(eb);
+            Function roundb = ff.function("round_2", eb);
+            Function hexb = ff.function("tohex", roundb);
+            rgb.add(hexb);
+
+            Expression[] args = new Expression[rgb.size()];
+            args = rgb.toArray(args);
+            Function concat = ff.function("Concatenate", args);
+            return concat;
+        } else {
+            throw new MBFormatException("Expression \"rgb\" requires exactly 3 arguments");
         }
+    }
 
     /**
      * Creates a color value from red, green, blue components, which must range between 0 and 255, and an alpha
      * component which must range between 0 and 1. If any component is out of range, the expression is an error.
      * Example: ["rgba", number, number, number, number]: color
      * @return
      */
+    // Currently unsupported
     public Expression colorRGBA(){
-        return null;
+            throw new MBFormatException("RGBA colors are currently unsupported");
```

Comment:
`UnsupportedOperationException` is more appropriate here, although I don't know if that breaks API?
Looks like similar code has been using `MBFormatException`, although that is more for things that aren't supported by the spec, so I would still prefer `UnsupportedOperationException` in this case.